iceoryx_binding_c/test/moduletests/test_chunk.cpp

```diff
@@ -159,4 +159,60 @@ TEST_F(Chunk_test, GettingUserHeaderFromConstChunkHeaderWorks)
     EXPECT_THAT(userHeaderStartAddress - chunkStartAddress, Eq(sizeof(ChunkHeader)));
 }
 
+TEST_F(Chunk_test, GettingHeaderPropertiesFromConstChunkHeaderWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0d561c7e-ae29-11ec-b909-0242ac120002");
+    constexpr uint32_t USER_PAYLOAD_SIZE(42U);
+    constexpr uint32_t USER_PAYLOAD_ALIGNMENT(64U);
+    constexpr uint32_t USER_HEADER_SIZE = 16U;
+    constexpr uint32_t USER_HEADER_ALIGNMENT = 8U;
+    void* userPayload{nullptr};
+    ASSERT_EQ(iox_pub_loan_aligned_chunk_with_user_header(publisher,
+                                                          &userPayload,
+                                                          USER_PAYLOAD_SIZE,
+                                                          USER_PAYLOAD_ALIGNMENT,
+                                                          USER_HEADER_SIZE,
+                                                          USER_HEADER_ALIGNMENT),
+              AllocationResult_SUCCESS);
+
+    const iox_chunk_header_t* chunkHeader = iox_chunk_header_from_user_payload(userPayload);
+
+    ASSERT_EQ(iox_chunk_header_user_chunk_size(chunkHeader), USER_HEADER_SIZE + USER_PAYLOAD_SIZE);
```

Comment:
This is not correct since one has to also add the size of the chunk-header. But this is also not sufficient since the user-payload alignment is larger than 8, we have to take the worst case scenario into account so this would be more something like `sizeof(ChunkHeader + USER_HEADER_SIZE + USER_PAYLOAD_ALIGNMENT - 8 + USER_PAYLOAD_SIZE)`. But this is also only the minimum required chunk size. Depending on the available mempools, the size of the actual loaned chunk most certainly will be larger.

Although it would be a whitebox test it might be better to just cast `chunkHeader` into a `iox::mepoo::ChunkHeader` and call the `chunkSize` method.

Reply:
> Although it would be a whitebox test it might be better to just cast chunkHeader into a iox::mepoo::ChunkHeader and call the chunkSize method.

I would definitely prefer this approach since the implementation should also use this method behind the scenes.